pokemongo_bot/path_navigator.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+import logger
+from cell_workers.utils import distance, i2f, format_dist
+from human_behaviour import sleep
+from step_walker import StepWalker
+from pgoapi.utilities import f2i
+
+
+class PathNavigator(object):
+    def __init__(self, bot):
+        self.bot = bot
+        self.api = bot.api
+        self.config = bot.config
+        self.ptr = 0
+        self.mode = self.get_mode()
+
+        if not self.config.path_navigator:
+            raise RuntimeError('path_navigator config missing')
+
+        self.points = self.get_points()
+
+    def get_points(self):
+        if not self.config.path_navigator.has_key('points') or len(self.config.path_navigator['points']) < 2:
+            raise RuntimeError('You need to specify at least two points in the path_navigator config')
+
+        points = [point.split(',') for point in self.config.path_navigator['points']]
+        return points
+
+    def get_mode(self):
+        mode = 'linear'
+        if self.config.path_navigator.has_key('mode'):
+            if self.config.path_navigator['mode'] == 'loop':
+                mode = 'loop'
+
+        return mode
+
+    def take_step(self):
+        point = self.points[self.ptr]
+        lat = float(point[0])
+        lng = float(point[1])
+
+        if self.config.walk > 0:
+            step_walker = StepWalker(
+                self.bot,
+                self.config.walk,
+                lat,
+                lng
+            )
+
+            if step_walker.step():
+                step_walker = None
```

Comment:
You don't need to set `step_walker` as `None` here, because the `StepWalker` is being recreated every time a step is taken.